controllers/orders.js

```diff
@@ -58,7 +58,43 @@ const psMinusWaivePoints = new PS({
 const psUpdateFoodCount = new PS({
     name: 'update-food-count',
     text: `UPDATE Food set numOrders = (numOrders + $3) WHERE fname = $1 and rid = $2`
-})
+});
+
+const psGetMonthlyOrderSummary = new PS({ name: 'monthly-order-summary',
+    text: "SELECT date_part('month', ordertime) as month, date_part('year', ordertime) as year, count(*) as order_count" +
+    " FROM orders " +
+    "GROUP BY month, year " +
+    "ORDER BY year, month;"});
+
+const psGetYearlyOrderSummary = new PS({ name: 'yearly-order-summary',
+    text: "SELECT date_part('year', ordertime) as year, count(*) as order_count" +
+    " FROM orders " +
+    "GROUP BY year " +
+    "ORDER BY year;"});
+
+const psGetYearlySalesSummary = new PS({ name: 'yearly-sales-summary',
+    text: "SELECT date_part('year', ordertime) as year, sum(finalprice) as yearly_sales" +
```

Comment:
i think we shld be counting completed orders for sales summary so shld use `delivertime`.

Reply:
If you  extract from null delivered time (incomplete orders), you'' get null year and month. checking for for non null should fix this. This should apply to similar queries